src/components/DateInput/DateInput.tsx

```diff
@@ -77,7 +85,19 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
   const targetRef = React.useRef(null);
   const [currentDate, setCurrentDate] = useState(dateToDayjs(value, timezone));
   const [currentMonth, setCurrentMonth] = useState(currentDate || now(timezone));
+  const resetLabel = React.useMemo(() => (withTime ? 'Clear Date & Time' : 'Clear Date'), [
+    withTime,
+  ]);
+
   const { isOpen, open, close } = useDisclosure();
+  const previousDate = usePrevious(dateToDayjs(value, timezone));
+
+  const isDisabled = React.useMemo(() => {
+    if (dayjs.isDayjs(currentDate) && dayjs.isDayjs(previousDate)) {
```

Comment:
wouldn't this check suffice?  This is because in my mind, `currentDate` is either a dayjs object or it's `undefined`

```suggestion
    if (currentDate) {
```

Reply:
`previousDate` can be undefined and `dayjs` is failing big.